hare/algorithm.go

```diff
@@ -537,11 +581,15 @@ func (proc *consensusProcess) onRoundEnd(ctx context.Context) {
 
 // advances the state to the next round.
 func (proc *consensusProcess) advanceToNextRound(ctx context.Context) {
-	k := proc.addToK(1)
-	if k >= 4 && k%4 == 0 {
-		proc.WithContext(ctx).Event().Warning("starting new iteration",
-			log.Uint32("current_k", k),
-			proc.instanceID)
+	if proc.completed && proc.currentRound() == notifyRound {
+		proc.k = certifyRound
```

Comment:
I've noticed that `proc.k` has accessors that use `atomic` operations (`addToK`, `getK` and `setK`) that you've stopped using. In my Hare round clock PR I've also used `k` directly. I think this might be a mistake and we should switch back to using only the accessors. If you think `k` can't be accessed concurrently then we should clean up and remove the accessors.